lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs

```diff
@@ -448,18 +450,17 @@ testMinSeverityFromEnv =
     minSeverityFromEnv Notice "TESTS_TRACING_MIN_SEVERITY"
 
 -- | Allow configuring which port the wallet server listen to in an integration
--- setup.
+-- setup. Crashes if the variable is not a number.
 walletListenFromEnv :: IO Listen
-walletListenFromEnv = lookupEnv "CARDANO_WALLET_PORT" >>= \case
+walletListenFromEnv = envFromText "CARDANO_WALLET_PORT" >>= \case
     Nothing -> pure ListenOnRandomPort
-    Just x  -> case fromText (T.pack x) of
-        Right port -> pure $ ListenOnPort port
-        Left e -> die (show e)
+    Just (Right port) -> pure $ ListenOnPort port
+    Just (Left e) -> die $ show e
 
 -- | Directory for extra logging. Buildkite will set this environment variable
 -- and upload logs in it automatically.
 testLogDirFromEnv :: IO (Maybe FilePath)
-testLogDirFromEnv = traverse makeAbsolute =<< lookupEnv "TESTS_LOGDIR"
+testLogDirFromEnv = traverse makeAbsolute =<< lookupEnvNonEmpty "TESTS_LOGDIR"
```

Comment:
`makeAbsolute ""` equals `pwd`. Isn't that acceptable?

Reply:
Not really - the normal tests log directory should be used when that variable is empty or unset.